acvm-repo/brillig_vm/src/lib.rs

```diff
@@ -196,6 +212,13 @@ impl<'a, F: AcirField, B: BlackBoxFunctionSolver<F>> VM<'a, F, B> {
 
     /// Process a single opcode and modify the program counter.
     pub fn process_opcode(&mut self) -> VMStatus<F> {
+        let call_stack: Vec<usize> = self.get_call_stack();
+        self.profiling_samples.push(BrilligProfilingSample { call_stack });
+
+        self.process_opcode_internal()
+    }
```

Comment:
Is it a concern that this profiling happens regardless of whether profiling is active? I don't know if this makes thing slower. But this is the first time I review a brillig_vm PR, so let me know if what I'm saying doesn't make sense.

Reply:
I don't think it should make things much slower, but it is hard to tell as we do not have comprehensive benchmarks for execution timings like we do bytecode sizes. I will change this either way to mark whether profiling is active and whether to collect samples.

Reply:
Yes It should be way better with the flag because if not then the process opcode codepath (the hottest) will trigger vec reallocations which are very expensive